### PR TITLE
Match file types using getSupportInfo from prettier

### PR DESCRIPTION
**Pull request by vlad-zhukov**
Just a starting point of #137

### CODE REVIEW

File: src/__tests__/utils.js [hunk 1]
```diff
@@ -1,216 +1,246 @@
...
+      expect(matchFileType('JavaScript', 'test.es')).toBe(true);
+      expect(matchFileType('CSS', 'test.css')).toBe(true);
+      expect(matchFileType('Less', 'test.less')).toBe(true);
+  });
 });
 
-// eslint-disable-next-line max-len
-test(`if fallbacks are provided, those are preferred over disabled eslint rules`, () => {
-  const { prettier } = getOptionsForFormatting(
+describe("getOptionsForFormatting", () => {
```
Comment (vlad-zhukov) [inline src/__tests__/utils.js]: Sorry for a mess in this file, just wrapped tests with a `describe` block.

File: src/utils.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import prettier from "prettier";
```
Comment (vlad-zhukov) [inline src/utils.js]: Should I import it like this or use `requireModule` from `index.js`?

**Comment by zimme** [comment]
Thank you for taking the time to do this. Really appreciate it. However I'm actually pretty much done with a solution for this that doesn't even have to check for filetype at all, we just forward `text` and `filePath` to `prettier` directly and then `prettier` takes care of figuring out which parser it should use.

Then the only thing we need to figure out is when to run prettier only. i.e. just check for filetypes that `eslint` supports. For now the solution just checks for `.js|jsx` and `.ts|tsx` files, but I want it to see if we can extend that to use `extensions` from `eslint` config instead when available.

**Comment by zimme** [comment]
I'll put up my solution as PR tomorrow as I don't think I'll have time tonight and then we can compare and see which solution we think is the "best" way forward. I want to try and make this package simpler, as it will make maintaining it easier.

File: src/__tests__/utils.js (hunk 1 above)
Comment (zimme) [reply src/__tests__/utils.js]: I think it sounds good to do this in a separate PR, we can work in a refactoring PR together once we've solved this problem.

File: src/utils.js (hunk 2 above)
Comment (zimme) [reply src/utils.js]: I'm thinking let's use `requireModule` and then we can come back to this. It's related to the discussion in #133.

**Comment by zimme** [comment]
Thanks for the effort and input. Closing this now as #145 was merged.

**Pull request closed without merging**